### PR TITLE
User agent header

### PR DESCRIPTION
**Pull request by mateuszmidor**
KNTK-411

- add User-Agent : kentik_community_sdk_python/{latest git tag} header in http requests
- add user-agent : kentik_community_sdk_python/{latest git tag}  metadata in grpc requests

### CODE REVIEW

File: kentik_api_library/setup.py [hunk 1]
```diff
@@ -239,6 +239,34 @@ def run(self):
         print(*packages, sep="\n")
 
 
+class SetupAPIClientVersion(Command):
```
Comment (mmac-m3a) [inline kentik_api_library/setup.py:242]: This is not needed. [setuptools_scm](https://github.com/pypa/setuptools_scm) can provide version.
You just need to add:
```
from setuptools_scm import get_version
```
and call `get_version()` where necessary.

Comment (mmac-m3a) [reply kentik_api_library/setup.py:242]: I've also found out that `setuptools_scm` older than 7.0 has a bug that prevents it from working in this project (and I'd say any other too :-(). We will need to add:
```
[build-system]
requires = ["setuptools_scm>=7.0", "wheel"]
```
to `pyproject.toml` ... and after that we can also remove the `relative_to` directive from the `[tool.setuptools_scm]` section (as you've proposed some time ago).

File: kentik_api_library/kentik_api/kentik_api.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ def __init__(
             logging.debug("KentikAPI: null api_host, setting to %s", self.API_HOST_US)
             api_host = self.API_HOST_US
         api_v5_url = self.make_api_v5_url(api_host)
-        connector = APIConnector(api_v5_url, auth_email, auth_token, timeout, retry_strategy, proxy)
+        connector = APIConnector(api_v5_url, auth_email, auth_token, client_version, timeout, retry_strategy, proxy)
```
Comment (mmac-m3a) [inline kentik_api_library/kentik_api/kentik_api.py:46]: Let's retrieve the version in `APIConnector` as opposed to passing it as an argument to the constructor.

File: kentik_api_library/kentik_api/api_connection/api_connector.py [hunk 3]
```diff
@@ -58,16 +60,16 @@ def send(self, api_call: APICall, payload: Optional[Dict[str, Any]] = None) -> A
         return APICallResponse(response.status_code, response.text)
 
     def _do_request(self, api_call: APICall, payload: Optional[Dict[str, Any]] = None) -> Response:
-
         url = self._get_api_query_url(api_call.url_path)
+        headers = {"User-Agent": self._client_version}
         if api_call.method == APICallMethods.GET:
-            response = self._session.get(url, params=payload, timeout=self._timeout)
+            response = self._session.get(url, params=payload, headers=headers, timeout=self._timeout)
```
Comment (mmac-m3a) [inline kentik_api_library/kentik_api/api_connection/api_connector.py:66]: Set `self._session.headers` in the constructor as opposed to passing it in every call.

File: .github/workflows/kentik-api.yml [hunk 4]
```diff
@@ -83,6 +83,7 @@ jobs:
           pip3 install -r requirements.txt
           pip3 install -r requirements-dev.txt
           python3 setup.py grpc_stubs # building grpc stubs must go before building
+          python3 setup.py setup_client_version # set sdk api client version to the latest git tag
```
Comment (mmac-m3a) [inline .github/workflows/kentik-api.yml:86]: `setuptools_scm` should do it automagically.

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
Please, use `setuptools_scm` to retrieve package version and also address my other comments.

File: kentik_api_library/setup.py (hunk 1 above)
Comment (mateuszmidor) [reply kentik_api_library/setup.py:242]: Updating

File: kentik_api_library/kentik_api/kentik_api.py (hunk 2 above)
Comment (mateuszmidor) [reply kentik_api_library/kentik_api/kentik_api.py:46]: Updating

File: kentik_api_library/kentik_api/api_connection/api_connector.py (hunk 3 above)
Comment (mateuszmidor) [reply kentik_api_library/kentik_api/api_connection/api_connector.py:66]: Updating

File: .github/workflows/kentik-api.yml (hunk 4 above)
Comment (mateuszmidor) [reply .github/workflows/kentik-api.yml:86]: This call updates the content of "version.py" file

**Review by danielfurman — COMMENTED** [review]
An automatic test for new headers would be nice to have

File: kentik_api_library/setup.py (hunk 1 above)
Comment (mmac-m3a) [reply kentik_api_library/setup.py:242]: You are still working on it, or  do you consider it done?

Comment (mateuszmidor) [reply kentik_api_library/setup.py:242]: Done. Or do you have further suggestions?

**Review by danielfurman — APPROVED** [review]

File: kentik_api_library/setup.py (hunk 1 above)
Comment (mmac-m3a) [reply kentik_api_library/setup.py:242]: Well, you've really not addressed my comments. I'll merge the code and address them myself.

**Review by mmac-m3a — APPROVED** [review]
OK, for now.

**Pull request merged**